tests/test_loading.py

```diff
@@ -1,36 +1,107 @@
+    form_path = FormFieldClassifier._form_filename(model_path)
+
+    assert not os.path.exists(model_path)
+    assert not os.path.exists(field_path)
+    assert not os.path.exists(form_path)
 
     with pytest.raises(IOError):
-        formasaurus.FormFieldClassifier.load(model, autocreate=False)
+        formasaurus.FormFieldClassifier.load(model_path, autocreate=False)
 
-    assert not os.path.exists(model)
+    assert not os.path.exists(model_path)
+    assert not os.path.exists(field_path)
+    assert not os.path.exists(form_path)
 
 
 def test_autocreate_model(tmpdir):
-    model = os.path.join(str(tmpdir), "m.joblib")
-    formasaurus.FormFieldClassifier.load(model)
-    assert os.path.exists(model)
+    model_path = os.path.join(str(tmpdir), FILENAME)
+    field_path = FormFieldClassifier._field_filename(model_path)
+    form_path = FormFieldClassifier._form_filename(model_path)
+
+    assert not os.path.exists(model_path)
+    assert not os.path.exists(field_path)
+    assert not os.path.exists(form_path)
+
+    formasaurus.FormFieldClassifier.load(model_path)
+
+    assert not os.path.exists(model_path)
+    assert os.path.exists(field_path)
+    assert os.path.exists(form_path)
 
 
 def test_rebuild_model(tmpdir):
-    path = os.path.join(str(tmpdir), "m.joblib")
-    formasaurus.FormFieldClassifier.load(path)
-    mtime1 = os.path.getmtime(path)
+    model_path = os.path.join(str(tmpdir), FILENAME)
+    field_path = FormFieldClassifier._field_filename(model_path)
+    form_path = FormFieldClassifier._form_filename(model_path)
+
+    formasaurus.FormFieldClassifier.load(model_path)
+    field_mtime1 = os.path.getmtime(field_path)
+    form_mtime1 = os.path.getmtime(form_path)
+
+    formasaurus.FormFieldClassifier.load(model_path)
+    field_mtime2 = os.path.getmtime(field_path)
+    form_mtime2 = os.path.getmtime(form_path)
+
+    formasaurus.FormFieldClassifier.load(model_path, rebuild=True)
+    field_mtime3 = os.path.getmtime(field_path)
+    form_mtime3 = os.path.getmtime(form_path)
+
+    assert field_mtime2 == field_mtime1
+    assert field_mtime3 > field_mtime1
+    assert form_mtime2 == form_mtime1
+    assert form_mtime3 > form_mtime1
+
+
+def test_save_and_load(tmpdir):
+    """Reloading a model from disk should not change the model."""
+    model_path = os.path.join(str(tmpdir), FILENAME)
+
+    a_start_time = process_time()
+
+    # Since model_path does not exist yet, this line trains a new model on the
+    # spot and saves it into model_path.
+    model_a = formasaurus.FormFieldClassifier.load(model_path)
+
+    b_start_time = process_time()
+    a_time = b_start_time - a_start_time
+
+    # Loads the model generated and saved above.
+    model_b = formasaurus.FormFieldClassifier.load(model_path)
+
+    b_time = process_time() - b_start_time
+    assert (
+        b_time / a_time < 0.01
+    ), "Unserializing takes too long, over 1% of the training time"
```

Comment:
This might be overkill, but I was not confident enough that I was avoiding training. In fact, I am not completely sure even now :sweat: